modules/ximgproc/include/opencv2/ximgproc/fourier_descriptors.hpp

```diff
@@ -0,0 +1,112 @@
+*  (3 - clause BSD License)
+*
+*  Redistribution and use in source and binary forms, with or without modification,
+*  are permitted provided that the following conditions are met :
+*
+*  * Redistributions of source code must retain the above copyright notice,
+*  this list of conditions and the following disclaimer.
+*
+*  * Redistributions in binary form must reproduce the above copyright notice,
+*  this list of conditions and the following disclaimer in the documentation
+*  and / or other materials provided with the distribution.
+*
+*  * Neither the names of the copyright holders nor the names of the contributors
+*  may be used to endorse or promote products derived from this software
+*  without specific prior written permission.
+*
+*  This software is provided by the copyright holders and contributors "as is" and
+*  any express or implied warranties, including, but not limited to, the implied
+*  warranties of merchantability and fitness for a particular purpose are disclaimed.
+*  In no event shall copyright holders or contributors be liable for any direct,
+*  indirect, incidental, special, exemplary, or consequential damages
+*  (including, but not limited to, procurement of substitute goods or services;
+*  loss of use, data, or profits; or business interruption) however caused
+*  and on any theory of liability, whether in contract, strict liability,
+*  or tort(including negligence or otherwise) arising in any way out of
+*  the use of this software, even if advised of the possibility of such damage.
+*/
+
+#ifndef __OPENCV_FOURIERDESCRIPTORS_HPP__
+#define __OPENCV_FOURIERDESCRIPTORS_HPP__
+#ifdef __cplusplus
+
+#include <opencv2/core.hpp>
+
+namespace cv {
+    namespace ximgproc {
+
+        //! @addtogroup shape
+        //! @{
+
+        /** @brief Abstract base class for shape transformation algorithms.
+        */
+        class CV_EXPORTS ContourFitting : public Algorithm
+        {
+            int ctrSize;
+            int fdSize;
+            std::vector<std::complex<double> > b;
+            std::vector<std::complex<double> > a;
+            std::vector<double> frequence;
+            std::vector<double> rho, psi;
+            void frequencyInit();
+            void fAlpha(double x, double &fn, double &df);
+            double distance(std::complex<double> r, double alpha);
+            double  newtonRaphson(double x1, double x2);
+        public:
+            /** @brief Fit two closed curves using fourier descriptors.
+
+            @param src Contour defining first shape.
+            @param dst Contour defining second shape (Target).
+            @param rotOrigScale Standard vector angle origin and scale.
+            @param dist distance between src and dst after matching.
+            */
+            ContourFitting(int ctr=1024,int fd=16):ctrSize(ctr),fdSize(fd){};
+            Mat estimateTransformation(InputArray src, InputArray dst, double *dist=0,bool fdContour=false);
```

Comment:
Why do you use `Mat` as return type here? Usually result is returned via `OutputArray`.